common/src/misc/utils.ts

```diff
@@ -203,7 +203,7 @@ export class Utils {
             return null;
         }
 
-        uriString = uriString.trim();
+        uriString = uriString.replace(/\$/g, '').trim();
         if (uriString === '') {
             return null;
         }
```

Comment:
Down below line 213 after the `data:` check. Add another  check `tldjs.isValid()` and if not valid, then return `null`. Follow the patterns for calling `tldjs` as also seen below.